src-js/optimizely.js

```diff
@@ -88,4 +100,76 @@ class OptimizelyClient {
   }
 }
 
+// trackExperimentViewed checks if we alredy have sent the Experiment Viewed
+// event to segment/amplitude by looking into the localstorage.
+// If not, it builds the properties needed and call trackAction with it
+const trackExperimentViewed = (orgId, experimentKey, experimentId, variationName, variationId, userId) => {
+  if (!isExperimentAlreadyViewed(orgId, experimentKey)) {
+    const properties = {
+      id: uuidv4(),
+      timestamp: new Date().toISOString(),
+      experimentId,
+      experimentName: experimentKey,
+      allocationType: 'organization_id',
+      orgId,
+      projectId: null, // This experiment is measured at the org level
+      userId,
+      variationId,
+      variationName,
+    };
+
+    // send event with the properly formatted properties
+    window.AnalyticsClient.trackAction('Experiment Viewed', properties);
+
+    // store experiment participation in localstorage
+    storeExperimentParticipation(orgId, experimentKey, variationName);
+  }
+}
+
+// isExperimentAlreadyViewed checks in the localstorage if we already
+// marked the experiment as viewed
+const isExperimentAlreadyViewed = (orgId, experimentKey) => {
+  try {
+    const experiments = JSON.parse(localStorage.getItem(STORAGE_KEY));
+    return (experiments && experiments.hasOwnProperty(orgId) && experiments[orgId] && experiments[orgId].hasOwnProperty(experimentKey));
+  } catch {
+    return false
+  }
+}
+
+// storeExperimentParticipation stores the experiment variationName in the
+// localstorage
+const storeExperimentParticipation = (orgId, experimentKey, variationName) => {
+  if (!orgId || !experimentKey || !variationName) {
+    return;
+  }
+
+  // get exepriments out of localstorage
+  let experiments;
+  try {
+    experiments = JSON.parse(localStorage.getItem(STORAGE_KEY)) ?? {};
+  } catch {
+    experiments = {};
+  }
+
+  // if we have nothing we start de build the `experiments` object
+  if (!experiments[orgId]) {
+    experiments[orgId] = {};
+  }
+  if (!experiments[orgId][experimentKey]) {
+    experiments[orgId][experimentKey] = {};
+  }
+
+  // assign the current experiment and its values
+  experiments[orgId][experimentKey] = { variationName, createdAt: new Date().getTime() };
+
+  try {
+    // setItem will sometimes fail with a "Quota Exceeded" exception if users have custom configurations
+    localStorage.setItem(STORAGE_KEY, JSON.stringify(experiments));
+  } catch {
+    // We're deliberately ignoring it so that it doesn't break the app. It'll mean a few extra
+    // events are emitted, but I think that's the lesser issue.
+  }
+}
+
 export default OptimizelyClient;
```

Comment:
May have to set preferences per repo? As per Tony: VSCode to add it: Code > Preferences > Settings